stellar-contract-env-host/src/test.rs

```diff
@@ -392,6 +394,100 @@ fn vec_append_empty() {
     assert_eq!(host.obj_cmp(obj0.into(), obj1.into()).unwrap(), 0);
 }
 
+/// crypto tests
+#[test]
+fn sha256_test() {
+    let host = Host::default();
+    let obj0 = host
+        .to_host_obj(&ScObject::Binary(vec![1].try_into().unwrap()))
+        .unwrap();
+    let hash_obj = host.compute_hash_sha256(obj0.to_object()).unwrap();
+
+    let v = host.from_host_val(hash_obj.to_raw()).unwrap();
+    let bin = match v {
+        ScVal::Object(Some(scobj)) => match scobj {
+            ScObject::Binary(bin) => bin,
+            _ => panic!("Wrong type"),
+        },
+        _ => panic!("Wrong type"),
+    };
+
+    /*
+    We took the sha256 of [1], which is 4bf5122f344554c53bde2ebb8cd2b7e3d1600ad631c385a5d7cce23c7785459a
+    The exp array contains the decimal representation of each hex value
+    */
+    let exp: Vec<u8> = vec![
+        75, 245, 18, 47, 52, 69, 84, 197, 59, 222, 46, 187, 140, 210, 183, 227, 209, 96, 10, 214,
+        49, 195, 133, 165, 215, 204, 226, 60, 119, 133, 69, 154,
+    ];
+    assert_eq!(bin.as_vec().clone(), exp);
+}
+
+fn is_true(val: ScVal) -> bool {
+    let res = match val {
+        ScVal::Static(ScStatic::True) => true,
+        ScVal::Static(ScStatic::False) => false,
+        _ => panic!("Wrong type"),
+    };
+
+    res
+}
+
+#[test]
+fn ed25519_verify_test() {
+    let host = Host::default();
+
+    // From https://datatracker.ietf.org/doc/html/rfc8032#section-7.1
+
+    // First verify successfully
+    let public_key: &[u8] = b"3d4017c3e843895a92b70aa74d1b7ebc9c982ccf2ec4968cc0cd55f12af4660c";
+    let message: &[u8] = b"72";
+    let signature: &[u8] = b"92a009a9f0d4cab8720e820b5f642540a2b27b5416503f8fb3762223ebdb69da085ac1e43e15996e458f3613d0f11d8c387b2eaeb4302aeeb00d291612bb0c00";
```

Comment:
Can we get a more realistic example, like an actual transaction going through the core, and extracting its signature. This way we can be more sure this library is behaving same as the core's implementation (they should, since both are ed25519). 
And I'm not suggesting to unit test the ed25519-dalek crate, but just one example that matches with core will give us more confidence.

Reply:
Yeah we should _definitely_ expose core crypto functions to cross-validation through the rust bridge, from rust tests (or vice versa -- but I suspect the rust test infrastructure is substantially richer and more pleasant than Catch2 and autocheck)

Reply:
Yeah that would be even better.
@graydon in that case, would we be better off using the core crypto functions directly? Is that possible?

Reply:
I'm not opposed to getting more realistic testing here, but I'm not going to hold up merging this to get that. Let's open an issue once this is merged to track improving the testing.

Reply:
I have some tests on the stellar-core side that test signature verification through `create_contract` as well. I agree we should add more tests, but I don't think that should hold up merging since these PR's get out of sync with master frequently due to the parallel work being done. I'll make sure to open an issue for testing like Jon mentioned if everyone agrees.